feature/experimental/bgp/ate_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,467 @@
+		bgpDut1.AddPeer().WithPeerAddress(peerIP).WithLocalASN(localAs).
+			WithHoldTime(holdTime)
+	}
+	return topo
+}
+
+// TestEstablish sets up and verifies basic BGP connection
+func TestEstablish(t *testing.T) {
+	// DUT configurations.
+	t.Logf("Start DUT config load:")
+	dut := ondatra.DUT(t, "dut")
+
+	// Configure interface on the DUT
+	t.Logf("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure BGP Neighbor on the DUT
+	t.Logf("Start DUT Network Instance and BGP Config")
+	configureNetworkInstance(t)
+
+	dutConfPath := dut.Config().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+	dutConfPath.Replace(t, nil)
+	dutConf := bgpCreateNbr(dutAS, ateAS, 0, "", "", holdTime0, ConnTime0, NegHoldTime0)
+	dutConfPath.Replace(t, dutConf)
+
+	// ATE Configuration.
+	t.Logf("Start ATE Config")
+	topo := configureATE(t, ateAS, dutAttrs.IPv4, "", 0, 0, "EXTERNAL")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	// Verify Port Status
+	t.Logf("Verifying port status")
+	verifyPortsUp(t, dut.Device)
+
+	t.Logf("Check BGP parameters")
+	verifyBgpTelemetry(t, dut)
+
+	t.Logf("Check BGP Capabilities")
+	verifyBGPCapabilities(t, dut)
+}
+
+// TestDisconnect is send cease notification from ate and verify reiceved error code on DUT
+func TestDisconnect(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	// Configure interface on the DUT
+	t.Logf("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	dutConfPath := dut.Config().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	statePath := dut.Telemetry().NetworkInstance(netInstance).Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+
+	// Configure BGP Neighbor on the DUT
+	t.Logf("Start DUT Network Instance and BGP Config")
+	configureNetworkInstance(t)
+	// Clear any existing config
+	fptest.LogYgot(t, "DUT BGP Config before", dutConfPath, dutConfPath.Get(t))
+	dutConfPath.Replace(t, nil)
+	dutConf := bgpCreateNbr(dutAS, ateAS, 0, "", "", holdTime0, ConnTime0, NegHoldTime0)
+	dutConfPath.Replace(t, dutConf)
+
+	t.Logf("configure port and BGP configs on ATE")
+	ate := ondatra.ATE(t, "ate")
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateAttrs.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateAttrs.IPv4CIDR()).WithDefaultGateway(dutAttrs.IPv4)
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutAttrs.IPv4).WithLocalASN(ateAS).WithTypeExternal()
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	t.Logf("Verify BGP session state : ESTABLISHED")
+	nbrPath.SessionState().Await(t, time.Second*100, telemetry.Bgp_Neighbor_SessionState_ESTABLISHED)
+
+	//TO DO: WAITING FOR THE FIX FROM ATE TO SEND NOTIFICATION TO DUT TO VERIFY ERROR CODE ON DUT
```

Comment:
I like to use if false to comment out code, since text editors will still format the code.

```
// TODO: waiting for ATE to support notification.
if false {
  t.Log("Send notification from ATE")
  // Add code here to send notification.
  
  t.Log("Verify BGP session state: ACTIVE")
  ...
}
```